include/graph.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <functional>
+
+#include "galois/Galois.h"
+#include "galois/graphs/MorphGraph.h"
+#include "galois/graphs/LS_LC_CSR_Graph.h"
+
+namespace scea {
+
+/* Contains the minimal functionalities that are benchmarked. */
+class Graph {
+public:
+  virtual int add_edges(uint64_t src, const std::vector<uint64_t> dsts) = 0;
+  virtual uint64_t size() noexcept                                      = 0;
+
+  virtual void for_each_edge(uint64_t src,
+                             std::function<void(uint64_t const&)> callback) = 0;
+};
```

Comment:
This needs to be titled `EditableGraph` or smthing like that. The `Graph` has the api as described in other places. I would also prefer use of `CRTP`.

Reply:
discussed this, putting CRTP off for now since it complicates the benchmarking code for x86